Fixed boolean field serialization.

stas opened the pull request:
I updated the Boolean field behavior to be as similar as possible with the de-serialization of the same field.

This breaks a test that verifies the field serialization of `None` to a `None`. I'm not sure what is the purpose of that, but it doesn't really make sense in the context of a boolean field.

Please help me understand if there's some backwards-compatibility or something with it, I will be happy to revert it back or fix the relevant test.

Thanks in advance for reviewing this.

Commit:
Fixed boolean field serialization.

Added docs and consistency checking for truthy values.

sloria commented:
All "simple" fields (`fields.Str`, `fields.Int`, etc.) serialize `None` to `None` if `allow_none=True`. I'm not sure that `fields.Boolean` should be an exception, though I can see the argument for symmetry with deserialization.

I'l have to think further about this. Feedback welcome.

stas commented:
Hey @sloria, I don't want this to be a blocker. I reverted back the `allow_none` behavior and added an issue to be able to track this concern and get more feedback #241.

Let me know if you need anything else in order to merge this.

stas commented:
@sloria I will be forever grateful if you could make some time and review this again. :bow:

sloria commented on marshmallow/fields.py:
```diff
@@ -607,6 +607,9 @@ class Boolean(Field):
     def _serialize(self, value, attr, obj):
         if value is None:
             return None
+        elif value in self.falsy:
```
For consistency, we should also check `if value in self.truthy` as well.

sloria replied:
Also, can you update the comments on ~L601 and ~L604 to reflect this change? Should just be a matter of changing "deserialize" to "(de)serialize".

sloria commented:
Just a few changes, and this should be good to merge. :+1: 

Also, don't forget to add yourself to AUTHORS.rst

stas commented:
Thanks @sloria 
I added your suggestions squashed and rebased the changes.

sloria commented:
Thanks!
